file_io.cpp

```diff
@@ -20,20 +22,44 @@
 #include "user_io.h"
 #include "cfg.h"
 #include "input.h"
+#include "miniz.h"
 
 int nDirEntries = 0;
 struct dirent DirItem[10000];
+const int maxDirEntries = (int)(sizeof(DirItem)/sizeof(DirItem[0]));
 int iSelectedEntry = 0;       // selected entry index
 int iFirstEntry = 0;
 
 static char full_path[1200];
 
+static bool FileIsZipped(const std::string path, std::string* zip_path, std::string* file_path)

```

Comment:
I used `std::string` to save a few lines of code and having to pass two extra length arguments here.  If you prefer I can rewrite this to use C strings and string functions.